crates/ra_mbe/src/lib.rs

```diff
@@ -50,6 +52,38 @@ pub(crate) struct Rule {
     pub(crate) rhs: tt::Subtree,
 }
 
+// Find the max token id inside a subtree
+fn max_id(subtree: &tt::Subtree) -> Option<u32> {
+    subtree
+        .token_trees
+        .iter()
+        .filter_map(|tt| match tt {
+            tt::TokenTree::Subtree(subtree) => max_id(subtree),
+            tt::TokenTree::Leaf(tt::Leaf::Ident(ident))
+                if ident.id != tt::TokenId::unspecified() =>
+            {
+                Some(ident.id.0)
+            }
+            _ => None,
```

Comment:
Maybe match all variants to make it impossible to forget changing this when adding a new variant.

Reply:
There are only 2 variants for this enum, `Subtree` and `Leaf`. The match all case `_` is for bypass the `if` condition. But I agree there should be a better way to write this code. :)